.github/actions/cdn_code_deploy/action.yaml

```diff
@@ -0,0 +1,100 @@
+name: CDN Code Deploy 
+description: An action Deploy the artifact to the CDN
+
+inputs:
+  inputs:
+    resource_group_name:
+      description: CDN resource group name.
+      type: string
+      required: true
+    storage_account_name:
+      description: Storage account name.
+      type: string
+      required: true
+    profile_cdn_name:
+      description: CDN Profile name.
+      type: string
+      required: true
+    endpoint_name:
+      description: CDN endpoint name.
+      type: string
+      required: true
+    blob_container_name:
+      description: Blob container name.
+      type: string
+      required: true
+    content_paths:
+      description: If you want to include only specific file set 'true' else 'false'
+      type: string
+      required: false
+    bundle_name:
+      description: Name of the bundle
+      type: string
+      required: false
+    arm_client_id:
+      description: ARM_CLIENT_ID
+      type: string
+      required: true
+    arm_tenant_id:
+      description: ARM_TENANT_ID
+      type: string
+      required: true
+    arm_subscription_id:
+      description: ARM_SUBSCRIPTION_ID
+      type: string
+      required: true
+runs:
+  using: composite
+  steps:
+    - name: Checkout
+      uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+    - name: Download Artifact
+      uses: actions/download-artifact@6b208ae046db98c579e8a3aa621ab581ff575935 # v4.1.1
+      with:
+        name: ${{ inputs.bundle_name || 'bundle' }}
+        path: "./${{ inputs.bundle_name || 'bundle' }}"
+
+    - name: Azure Login
+      uses: azure/login@v2 # v2.0.0
+      env:
+        ARM_USE_OIDC: true
+      with:
+        client-id: ${{ inputs.arm_client_id }}
+        tenant-id: ${{ inputs.arm_tenant_id }}
+        subscription-id: ${{ inputs.arm_subscription_id }}
+
+    - name: Sync Storage
+      shell: bash
+      run: |
+        az storage blob sync \
+          -c ${{ inputs.blob_container_name }} \
+          --account-name ${{ inputs.storage_account_name }} \
+          -s "./${{ inputs.bundle_name || 'bundle' }}"
+
+    - name: Found modified files
```

Comment:
what do you mean with 'modified' here?

Reply:
that compared to the previous commit the files are the same, but they have some changes

Reply:
what do you mean with "previous (commit)" here?

Reply:
the action that checks the changes sees them between the current commit of the branch in use and the last one of the main, and if we are already in the main it checks the last one of the main and the second to last one